osbenchmark/builder/downloaders/repositories/opensearch_distribution_repository_provider.py

```diff
@@ -0,0 +1,34 @@
+import logging
+
+from osbenchmark.builder.downloaders.repositories.repository_url_provider import RepositoryUrlProvider
+from osbenchmark.utils import convert
+
+
+class OpenSearchDistributionRepositoryProvider:
```

Comment:
What are your considerations for using ``RepositoryUrlProvider`` as a member instead of having a base class like ``RepositoryProvider`` ?

Reply:
Having `RepositoryUrlProvider` as a member variable rather than an abstract base class follows the design principle of preferring composition over inheritance. Rather than forcing logic inheritance from a base class, the RepoProviders can pick and choose which functionality they need through member variables